az-digital/az_quickstart
# Fixes #537 Relative paths in Cards link field

This pull request adds the ability for cards to use internal paths in the link field.

## Description
This pull request adds the ability for cards to use internal paths in the link field. This requires using a custom validation function because the Form API `url` element does not support internal paths.

Core uses an autocomplete field (that finds entities by titles) for internal paths and we may want to consider this in the future for general availability.

The Card Formatter has had its style of dependency injection updated because the path validator service was being added to it.

**This PR is based on my other card branch (not main) and that PR should be merged first. This PR is fairly small; the large delta is because it is based on `feature/427` and not `main`.**

Consequently, the other card changes will also appear in this branch.

## Related Issue
#537 

## How Has This Been Tested?

- Create a page
- Create cards
- Add links for cards.
- Verify that external links and internal links both work (eg `<front>`, `/node/2/`. `http://www.google.com`) all continue to work.
- Verify that linking to an internal path that doesn't exist produces an error message.
- Create a link to a privileged page, e.g. `/admin/content` and verify that when the card is saved, the link is not rendered for an anonymous user looking at the page.


## Types of changes


## Checklist:
